src/query.c

```diff
@@ -1818,13 +1815,14 @@ static int QueryNode_CheckIsValid(QueryNode *n, IndexSpec *spec, RSSearchOptions
       break;
   }
 
+  // Check for weight attribute
+  if (res == REDISMODULE_OK && (validationFlags & QAST_NO_WEIGHT_ATTRIBUTE) && n->opts.explicitWeight) {
+    QueryError_SetError(status, QUERY_ESYNTAX, "Weight attribute is not allowed");
```

Comment:
I think the previous error message is better.. Consider having the flag be for vector filter validation, such that we can check for the same flag and raise the same error here (as done above for the Vector part).
Let's add the error to `QUERY_XERRS` in `query_error.h`.

Reply:
OK. Now the error messages are in `QUERY_XERRS`, we have 3 different error messages.
The tests were modified to check the validation and the obtained error message.